### PR TITLE
feat: Add helper function for generating multiple prekeys

### LINKED ISSUE #7: helper method to generate N prekeys
Clients need to generate many prekeys, other implementations have a helper method, e.g. Java has:

`
List<PreKeyRecord> preKeys         = KeyHelper.generatePreKeys(startId, 100);
`

### PR DESCRIPTION
**Pull request by pierwill**
Closes #7.

### CODE REVIEW

**Review by redshiftzero — CHANGES_REQUESTED** [review]
nice work! just one comment inline to look at

File: src/state.rs [hunk 1]
```diff
@@ -146,6 +147,28 @@ impl PreKeyRecord {
...
+/// from signal_protocol import curve, state
+/// prekeypair = curve.KeyPair.generate()
+/// prekeyid = 1
+/// manykeys = state.generate_n_prekeys(100, prekeyid, prekeypair)  # generates 100 keys
+/// ```
+#[pyfunction]
+pub fn generate_n_prekeys(n: u16, id: PreKeyId, keypair: &KeyPair) -> Vec<PreKeyRecord> {
+    let mut keyvec: Vec<PreKeyRecord> = Vec::new();
+    for _i in 0..n {
+        let prekey = PreKeyRecord::new(id, keypair);
```
[the remark this answers is not in the document]
Comment (pierwill) [reply src/state.rs:165]: Ah! I had wondered about this at some point. I'm on it.

[the remark this answers is not in the document]
Comment (pierwill) [reply src/state.rs:165]: Do we need to worry about PreKeyId collisions? Do we know (or need to know) globally which IDs have been used?

[the remark this answers is not in the document]
Comment (redshiftzero) [reply src/state.rs:165]: good question, my thinking is that the expectation should be that the PreKeyRecords returned from this function all have unique IDs (i.e. there are no collisions). 

Regarding used prekey IDs more generally, the object in these bindings that has that information is the `InMemSignalProtocolStore` (via the `InMemPreKeyStore` in the [upstream crate](https://github.com/signalapp/libsignal-client/blob/master/rust/protocol/src/storage/inmem.rs#L119-L128)). Right now as you can see from the linked logic, the prekey store will just transparently replace records with duplicated key IDs (imho it should probably be an error), but until that changes it's the responsibility of clients to (separately from this function) keep track of what their last used PreKeyID is so they don't accidentally overwrite a stored PreKeyRecord. 

let me know what you think!

[the remark this answers is not in the document]
Comment (pierwill) [reply src/state.rs]: Hm. A check for each ID into the InMemSignalProtocolStore (if it exists?) when generated is probably the most foolproof and robust way at the moment, no? :thinking:

File: src/state.rs [hunk 2]
```diff
@@ -146,6 +147,31 @@ impl PreKeyRecord {
...
+/// prekeypair = curve.KeyPair.generate()
+/// prekeyid = 1
+/// manykeys = state.generate_n_prekeys(100, prekeyid, prekeypair)  # generates 100 keys
+/// ```
+#[pyfunction]
+pub fn generate_n_prekeys(n: u16, id: PreKeyId) -> Vec<PreKeyRecord> {
+    let mut keyvec: Vec<PreKeyRecord> = Vec::new();
+    let mut i: u32 = id;
+    for _n in 0..n {
+        let keypair = KeyPair::generate();
```
Comment (pierwill) [inline src/state.rs:166]: When I play with this on the Python end, it still looks suspiciously like we're not getting a new KeyPair:

```python
In [2]: x = 1

In [3]: while x < 50:
   ...:     print(manykeys[x].id())
   ...:     print(manykeys[x].key_pair())
   ...:     x += 1
   ...: 
2
<KeyPair object at 0x733690406c38>
3
<KeyPair object at 0x733690406c38>
4
<KeyPair object at 0x733690406c38>
5
<KeyPair object at 0x733690406c38>
6
<KeyPair object at 0x733690406c38>
7
```

File: src/state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierwill) [reply src/state.rs]: Using `InMemSignalProtocolStore::get_pre_key`?

File: src/state.rs (hunk 2 above)
Comment (pierwill) [reply src/state.rs:166]: *Howeeever*, the test I wrote just passed.

```python
prekeyrecords = state.generate_n_prekeys(256, pre_key_id)
    assert len(prekeyrecords) == 256
    assert prekeyrecords[12].id() != prekeyrecords[13].id()
    assert prekeyrecords[14].key_pair() != prekeyrecords[15].key_pair()
```

File: src/state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierwill) [reply src/state.rs]: Although, reading your comments again, doing a collision check in the body of `state::generate_n_keys` might not be necessary given then implicit contract.

**Review by redshiftzero — APPROVED** [review]
looks good, thanks!

**Pull request merged**